README.md

```diff
@@ -1,3 +1,127 @@
-# orm-find-or-create-by
+# ORM: Preventing Record Duplication
 
-now that we have a way to save records and update records, what happens when we create a new ruby object that reflects a record that is already persisted to our database? In this case we want to update the original record. So, find_or_create_by
+## Objectives
+
+1. Learn how to avoid creating duplicate records in a database that is mapped to a Ruby program. 
+2. Build a `#find_or_create_by` method
+
+## The Dreaded Duplication
+
+What happens when two Ruby objects get created using the same attributes? If we are trying to persist representations of such objects to a database, would we end up with essentially identical rows in our table? That would make for a very confusing database and our program would quickly become useless as a way to store and manage information. 
+
+For example, lets say we have a `Song` class that produces individual song objects, each of which have a `name` and `album` attribute.
+
+Nothing stops us from creating two objects, each of which have the exact same name and album. 
+
+```ruby
+hello = Song.new("Hello", "25")
+hello_again = Song.new("Hello", "25")
+```
+
+What happens when we save these objects to our database?
+
+*For this example, we'll assume our connection to the database is stored in `DB[:conn]`*
+
+```ruby
+hello.save
+hello_again.save
+
+DB[:conn].execute("SELECT * FROM songs WHERE name = "Hello")
+# => [[1, "Hello", "25"], [2, "Hello", "25"]]
+```
+
+We have two records that contain the same information! How can we avoid this? When we try to save a new `Song` instance, we should first check to see the object we are trying to save already has an equivalent record in the database, if it does, we should simply update it, otherwise, we can go ahead and save it. 
+
+## Saving vs. Updating
+
+Let's say we have a song, `hello`:
+
+```ruby
+hello = Song.new("Hello", "25")
+```
+
+Before we call `#save` on our `hello` object, we need to check and see if a record containing this name and album already exists in the database. The SQL statement to accomplish that would look something like this:
+
+```ruby
+SELECT * FROM songs
+WHERE name = "Hello", album = "25";
+```
+
+If this statement returns a record, we don't need to create a new record, only update the existing one. Otherwise, we need to insert a new record into our database table. 
+
+Let's build a method that will allow us to either *find and update* and existing record or *create and save* and new one. 
```

Comment:
you never build this method? The find and update method isn't that useful. Are you trying to do the save thing where it updates if it's an existing record, and inserts if it's a new one? I think we cover that in the previous reading.  Either way to check to see if an object has persisted we do check to see if the `id` attribute is set, not a `find_by_name`